doc: matter: thread/wifi switching edits (#10225)
nrfconnect/sdk-nrf: samples/matter/lock/README.rst

You can use this sample as a reference for creating your application.

This device works as a Matter accessory device, meaning it can be paired and controlled remotely over a Matter network built on top of a low-power 802.15.4 Thread or Wi-Fi network.

> Support for both Thread and Wi-Fi depends on the hardware platform, so only one protocol at a time can be supported for a specific lock device.

Reviewer: I would add one additional sentence here, maybe:

```suggestion
Support for both Thread and Wi-Fi depends on the hardware platform. The Door Lock sample can be configured to utilize only one transport protocol e.g. Thread or Wi-Fi, or can use `Switchable Matter over Wi-Fi and Matter over Thread` architecture where application is able to load and run image with either Thread or Wi-Fi support on boot.
```

Author: Rephrased slightly, but added.

.. matter_door_lock_sample_remote_testing_end

.. _matter_lock_sample_wifi_thread_switching:

Thread and Wi-Fi switching
==========================

> When built using the ``thread_wifi_switched`` build type and programmed to the nRF5340 DK with the nRF7002 EK shield attached, the door lock sample supports a feature that allows you to dynamically switch between Matter over Thread and Matter over Wi-Fi.

Reviewer: ```suggestion
When built using the ``thread_wifi_switched`` build type and programmed to the nRF5340 DK with the nRF7002 EK shield attached, the door lock sample supports a feature that allows you to dynamically switch between Matter over Thread and Matter over Wi-Fi on device boot.
```

.. matter_door_lock_sample_thread_wifi_switch_desc_start

> The application is built in two variants: the first one is the application working with Matter over Thread and the second one is the application working with Matter over Wi-Fi.

Reviewer: ```suggestion
The application is built in two variants: the first one is the application working with Matter over Thread and the second one is the application working with Matter over Wi-Fi. It is configurable which transport is used by default (as a first one).
```

Both application variants are programmed into separate partitions of the external flash.
The application runs from the internal flash memory, using one of the variants from the external flash.
You can trigger the switch from one variant to another using the **Button 3** on the nRF5340 DK.

> The device is rebooted into the MCUboot bootloader, which swaps the current variant by swapping the application variant in the internal flash.

Reviewer 2: Swap word repeating
```suggestion
The device is rebooted into the MCUboot bootloader, which replaces the current variant by swapping the application variant in the internal flash.
```

> 

Reviewer: Could we add a note, that because external flash is used for both DFU and switching purpose, memory size requirements are different. Specifically it is required to use external flash with at least 6MB of memory.
